skfda/ml/classification/centroid_classifiers.py

```diff
@@ -0,0 +1,169 @@
+            of all samples that belong to that particular class are minimized.
+            By default it is used the usual mean, which minimizes the sum of L2
+            distances. This parameter allows change the centroid constructor.
+            The function must accept a :class:`FData` with the samples of one
+            class and return a :class:`FData` object with only one sample
+            representing the centroid.
+    Attributes:
+        centroids_: :class:`FDataGrid`
+            FDatagrid containing the centroid of each class
+    Examples:
+        Firstly, we will create a toy dataset with 2 classes
+
+        >>> from skfda.datasets import make_sinusoidal_process
+        >>> fd1 = make_sinusoidal_process(phase_std=.25, random_state=0)
+        >>> fd2 = make_sinusoidal_process(phase_mean=1.8, error_std=0.,
+        ...                               phase_std=.25, random_state=0)
+        >>> fd = fd1.concatenate(fd2)
+        >>> y = 15*[0] + 15*[1]
+
+        We will fit a Nearest centroids classifier
+
+        >>> from skfda.ml.classification import NearestCentroid
+        >>> neigh = NearestCentroid()
+        >>> neigh.fit(fd, y)
+        NearestCentroid(...)
+
+        We can predict the class of new samples
+
+        >>> neigh.predict(fd[::2]) # Predict labels for even samples
+        array([0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1])
+
+    See also:
+        :class:`~skfda.ml.classification.DTMClassifier
+    """
+
+    def __init__(self, metric=l2_distance, centroid=mean):
+        """Initialize the classifier."""
+        self.metric = metric
+        self.centroid = centroid
+
+    def fit(self, X, y):
+        """Fit the model using X as training data and y as target values.
+
+        Args:
+            X (:class:`FDataGrid`, array_matrix): Training data. FDataGrid
+                with the training data or array matrix with shape
+                [n_samples, n_samples] if metric='precomputed'.
+            y (array-like or sparse matrix): Target values of
+                shape = [n_samples] or [n_samples, n_outputs].
+        """
+        self.classes_, y_ind = _classifier_get_classes(y)
+
+        self.centroids_ = self.centroid(X[y_ind == 0])
+
+        for cur_class in range(1, self.classes_.size):
+            centroid = self.centroid(X[y_ind == cur_class])
+            self.centroids_ = self.centroids_.concatenate(centroid)
+
+        return self
+
+    def predict(self, X):
+        """Predict the class labels for the provided data.
+
+        Args:
+            X (:class:`FDataGrid`): FDataGrid with the test samples.
+
+        Returns:
+            y (np.array): array of shape [n_samples] or
+            [n_samples, n_outputs] with class labels for each data sample.
+        """
+        sklearn_check_is_fitted(self)
+
+        return self.classes_[pairwise_distance(self.metric)(
+            X, self.centroids_).argmin(axis=1)]
+
+
+class DTMClassifier(NearestCentroid):
```

Comment:
This is also in other PR. Please split the changes and tell me the intended order of the PRs, as there are dependencies between them.

Reply:
This branch was created based on the feature/DTM branch. I thought it was OK to do such things...

Reply:
Maybe it's best to wait until feature/DTM can be merged to merge this branch

Reply:
It is ok, I think. I just need to be aware, in order to merge the other before.